### PR TITLE
Improve Archival logic in TimerQueueProcessor

### PR DESCRIPTION
**Pull request by yycptt**
Also only execute the deleteHistory activity as local activity.

### CODE REVIEW

File: service/worker/archiver/activities.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func uploadHistoryActivity(ctx context.Context, request ArchiveRequest) (err err
 		logger.Error(carchiver.ArchiveNonRetriableErrorMsg, tag.ArchivalArchiveFailReason("failed to get history archiver"), tag.Error(err))
 		return errUploadNonRetriable
 	}
-	err = historyArchiver.Archive(ctx, URI, &carchiver.ArchiveHistoryRequest{
+	return historyArchiver.Archive(ctx, URI, &carchiver.ArchiveHistoryRequest{
```
Comment (yycptt) [inline service/worker/archiver/activities.go]: @andrewjdawson2016 Sorry I made you add the log message. I thought about it, it should be the archiver's responsibility to do the logging.

File: service/worker/archiver/activities.go [hunk 2]
```diff
@@ -82,15 +82,6 @@ func uploadHistoryActivity(ctx context.Context, request ArchiveRequest) (err err
 		NextEventID:          request.NextEventID,
 		CloseFailoverVersion: request.CloseFailoverVersion,
 	}, carchiver.GetHeartbeatArchiveOption(), carchiver.GetNonRetriableErrorOption(errUploadNonRetriable))
-	if err == nil {
```
Comment (andrewjdawson2016) [inline service/worker/archiver/activities.go]: I think we want this code?

File: service/worker/archiver/client.go [hunk 3]
```diff
@@ -46,6 +47,11 @@ type (
 		ArchiveInline  bool
 	}
 
+	// ClientResponse is the archive response returned from the archiver client
+	ClientResponse struct {
+		ArchivedInline bool
```
Comment (andrewjdawson2016) [inline service/worker/archiver/client.go]: On the ClientRequest ArchiveInline should be called AttemptArchiveInline

File: service/worker/archiver/handler.go [hunk 4]
```diff
@@ -135,31 +135,12 @@ func handleRequest(ctx workflow.Context, logger log.Logger, metricsClient metric
...
-			NonRetriableErrorReasons: deleteHistoryActivityNonRetryableErrors,
-		},
-	}
-	actCtx = workflow.WithActivityOptions(ctx, ao)
-	if err := workflow.ExecuteActivity(actCtx, deleteHistoryActivityFnName, request).Get(actCtx, nil); err != nil {
-		logger.Error("failed to delete history, this means zombie histories are left", tag.Error(err))
-		metricsClient.IncCounter(metrics.ArchiverScope, metrics.ArchiverDeleteFailedAllRetriesCount)
+	if err != nil {
+		logger.Error("deleting history though local activity failed, this means zombie histories are left", tag.Error(err))
+		metricsClient.IncCounter(metrics.ArchiverScope, metrics.ArchiverDeleteLocalFailedAllRetriesCount)
```
Comment (andrewjdawson2016) [inline service/worker/archiver/handler.go]: Can we change the metric name to just delete now that we are only trying to delete local?

File: service/worker/archiver/handler.go [hunk 5]
```diff
@@ -135,31 +135,12 @@ func handleRequest(ctx workflow.Context, logger log.Logger, metricsClient metric
...
-			ExpirationInterval:       5 * time.Minute,
-			NonRetriableErrorReasons: deleteHistoryActivityNonRetryableErrors,
-		},
-	}
-	actCtx = workflow.WithActivityOptions(ctx, ao)
-	if err := workflow.ExecuteActivity(actCtx, deleteHistoryActivityFnName, request).Get(actCtx, nil); err != nil {
-		logger.Error("failed to delete history, this means zombie histories are left", tag.Error(err))
-		metricsClient.IncCounter(metrics.ArchiverScope, metrics.ArchiverDeleteFailedAllRetriesCount)
+	if err != nil {
+		logger.Error("deleting history though local activity failed, this means zombie histories are left", tag.Error(err))
```
Comment (andrewjdawson2016) [inline service/worker/archiver/handler.go]: Can we remove the word local from the log. It suggests that we might try to delete some other way.

File: service/worker/archiver/activities.go (hunk 2 above)
Comment (yycptt) [reply service/worker/archiver/activities.go]: Added them back. Idea is it's ok to log in both the activity and in archiver.

**Pull request merged**